pkg/controller/job/job_controller.go

```diff
@@ -1364,11 +1380,21 @@ func jobSuspended(job *batch.Job) bool {
 	return job.Spec.Suspend != nil && *job.Spec.Suspend
 }
 
+// Recreate pods once they have fully transitioned to failed
+// This means that terminating pods will wait until they are fully deleted.
+func recreatePodsWhenFailed(job *batch.Job) bool {
+	if !feature.DefaultFeatureGate.Enabled(features.JobPodReplacementPolicy) {
+		return false
+	}
+	failed := batch.Failed
+	return pointer.StringEqual((*string)(&failed), (*string)(job.Spec.PodReplacementPolicy))
```

Comment:
much more readable to just use a conditional

Reply:
You are right.  Updated.